go/loader.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const defaultConfigFile = "./config.json"
+
+// getBoolEnv returns the bool value for an env var and a confirmation
+// if the var exists
+func getBoolEnv(name string) (bool, bool) {
+	if val := os.Getenv(name); val != "" {
+		return val == "true", true
+	}
+
+	return false, false
+}
+
+// getStringEnv returns the string value for an env var and a confirmation
+// if the var exists
+func getStringEnv(name string) (string, bool) {
+	if val := os.Getenv(name); val != "" {
+		return val, true
+	}
+
+	return "", false
+}
+
+// loadFromFile loads the agent config from a file
+func loadFromFile(c *AgentConfig, filename string) error {
```

Comment:
Can we make this more flexible so we also support yaml?  We can just do something cheesy based on filename extension.

Reply:
Yes we can support Yaml. I thought it was easier if we first landed JSON as it is in the standard library but I can try it.